src/gui/executionlog.cpp

```diff
@@ -35,18 +35,22 @@
 #include <QPalette>
 #include "executionlog.h"
 #include "ui_executionlog.h"
-#include "base/logger.h"
+#include "base/preferences.h"
 #include "guiiconprovider.h"
 #include "loglistwidget.h"
 
 ExecutionLog::ExecutionLog(QWidget *parent)
     : QWidget(parent)
     , ui(new Ui::ExecutionLog)
-    , m_msgList(new LogListWidget(MAX_LOG_MESSAGES))
-    , m_peerList(new LogListWidget(MAX_LOG_MESSAGES))
+    , m_peerList(new LogListWidget(MAX_LOG_MESSAGES, true, true, true, true))
```

Comment:
Exactly. The only "optimization" I could do is move the 2 formatting functions from `ExecutionLog` into the `LogListWidget` class as slots. Then in `ExecutionLog` connect the correct instance to the correct singal/slot.
So it is a tradeoff: A simple `ExecutionLog` or a simple `LogListWidget`. I don't really care which.